### PR TITLE
Utility improvements

### PR DESCRIPTION
**Pull request by krocard**
- asString now return the formated string leading to usage improvement
 - Fully unit test Utility.h
 - Simplify appendTitle implementation
 - CommandHandler: simplify the message formating
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/275%23issuecomment-146876912%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/275%23issuecomment-146939314%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/275%23issuecomment-146939406%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/275%23issuecomment-147371274%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/275%23issuecomment-147453762%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/275%23discussion_r41637913%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/275%23discussion_r41638387%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/275%23discussion_r41640883%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/275%23discussion_r41655589%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/275%23discussion_r41655599%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/275%23issuecomment-147698004%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/275%23issuecomment-146876912%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%20%40dawagner%20please%20review%22%2C%20%22created_at%22%3A%20%222015-10-09T13%3A51%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%5B100%25%20coverage%5D%28https%3A//coveralls.io/builds/3799144/source%3Ffilename%3Dutility%252FUtility.cpp%29%20on%20Utility.cpp%20%21%20%22%2C%20%22created_at%22%3A%20%222015-10-09T17%3A27%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-10-09T17%3A28%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40tcahuzax%20please%20review%22%2C%20%22created_at%22%3A%20%222015-10-12T11%3A37%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-10-12T16%3A32%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40tcahuzax%20please%20review%22%2C%20%22created_at%22%3A%20%222015-10-13T12%3A14%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20cdf01e0dd1dc43c0728e304016e1ac129fc55bad%20parameter/ConfigurableDomain.cpp%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/275%23discussion_r41637913%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20is%20has%20a%20completely%20different%20effect%20than%20it%20previously%20had.%20Same%20comment%20in%20all%20places%20below.%22%2C%20%22created_at%22%3A%20%222015-10-09T14%3A47%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22True%2C%20it%20is%20completly%20wrong.%22%2C%20%22created_at%22%3A%20%222015-10-09T15%3A10%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-10-09T17%3A25%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ConfigurableDomain.cpp%3AL301-308%22%7D%2C%20%22Pull%20cdf01e0dd1dc43c0728e304016e1ac129fc55bad%20utility/CMakeLists.txt%2014%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/275%23discussion_r41638387%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%27m%20not%20sure%20this%20is%20needed%20in%20this%20particular%20case.%22%2C%20%22created_at%22%3A%20%222015-10-09T14%3A51%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-10-09T17%3A25%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/CMakeLists.txt%3AL52-67%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/dawagner%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%7D%2C%20%22https%3A//github.com/krocard%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/dawagner'></a><a href='https://github.com/krocard'></a>

- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/275#issuecomment-146876912'>General Comment</a></b>
- <a href='https://github.com/krocard'></a> :+1: @dawagner please review
- <a href='https://github.com/krocard'></a> [100% coverage](https://coveralls.io/builds/3799144/source?filename=utility%2FUtility.cpp) on Utility.cpp !
- <a href='https://github.com/krocard'></a> :shipit: :+1:
- <a href='https://github.com/krocard'></a> @tcahuzax please review
- <a href='https://github.com/dawagner'></a> :shipit:
- <a href='https://github.com/krocard'></a> @tcahuzax please review
- [x] <a href='#crh-comment-Pull cdf01e0dd1dc43c0728e304016e1ac129fc55bad parameter/ConfigurableDomain.cpp 5'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/275#discussion_r41637913'>File: parameter/ConfigurableDomain.cpp:L301-308</a></b>
- <a href='https://github.com/dawagner'></a> This is has a completely different effect than it previously had. Same comment in all places below.
- <a href='https://github.com/krocard'></a> True, it is completly wrong.
- [x] <a href='#crh-comment-Pull cdf01e0dd1dc43c0728e304016e1ac129fc55bad utility/CMakeLists.txt 14'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/275#discussion_r41638387'>File: utility/CMakeLists.txt:L52-67</a></b>
- <a href='https://github.com/dawagner'></a> I'm not sure this is needed in this particular case.

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/275?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/275?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/275?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/275'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/275'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by krocard** [comment]
:+1: @dawagner please review

File: parameter/ConfigurableDomain.cpp [hunk 1]
```diff
@@ -301,7 +301,7 @@ bool CConfigurableDomain::parseConfigurableElements(const CXmlElement& xmlElemen
         core::Results infos;
         if (!addConfigurableElement(pConfigurableElement, NULL, infos)) {
 
-            CUtility::asString(infos, strError);
+            utility::asString(infos, strError);
```
Comment (dawagner) [inline parameter/ConfigurableDomain.cpp]: This is has a completely different effect than it previously had. Same comment in all places below.

File: utility/CMakeLists.txt [hunk 2]
```diff
@@ -52,3 +52,15 @@ install(FILES
...
+if(BUILD_TESTING)
+    # Add unit test
+    add_executable(utilityUnitTest test/utility.cpp)
+
+    target_link_libraries(utilityUnitTest pfw_utility catch)
+    add_test(NAME utilityUnitTest
+             COMMAND utilityUnitTest)
+
+    # Custom function defined in the top-level CMakeLists
+    set_test_env(utilityUnitTest)
```
Comment (dawagner) [inline utility/CMakeLists.txt]: I'm not sure this is needed in this particular case.

File: parameter/ConfigurableDomain.cpp (hunk 1 above)
Comment (krocard) [inline parameter/ConfigurableDomain.cpp]: True, it is completly wrong.

Comment (krocard) [inline parameter/ConfigurableDomain.cpp]: :+1:

File: utility/CMakeLists.txt (hunk 2 above)
Comment (krocard) [inline utility/CMakeLists.txt]: :+1:

**Comment by krocard** [comment]
[100% coverage](https://coveralls.io/builds/3799144/source?filename=utility%2FUtility.cpp) on Utility.cpp !

**Comment by krocard** [comment]
:shipit: :+1:

**Comment by krocard** [comment]
@tcahuzax please review

**Comment by dawagner** [comment]
:shipit:

**Comment by krocard** [comment]
@tcahuzax please review

**Comment by tcahuzax** [comment]
:+1:

**Pull request merged**